pkg/api/releases.go

```diff
@@ -43,31 +51,155 @@ func PrintPullRequestsReport(w http.ResponseWriter, req *http.Request, dbClient
 	RespondWithJSON(http.StatusOK, w, prs)
 }
 
-func PrintReleaseJobRunsReport(w http.ResponseWriter, req *http.Request, dbClient *db.DB) {
-	if dbClient == nil || dbClient.DB == nil {
-		RespondWithJSON(http.StatusOK, w, []struct{}{})
+func ListPayloadJobRuns(dbClient *db.DB, filterOpts *filter.FilterOptions, release string) ([]models.ReleaseJobRun, error) {
+	jobRuns := make([]models.ReleaseJobRun, 0)
+	var err error
+	q := dbClient.DB
+	if release != "" {
+		q = q.Where("release = ?", release)
 	}
-
-	q := releaseFilter(req, dbClient.DB)
 	q = q.Joins(`JOIN release_tags on release_tags.id = release_job_runs.release_tag_id`)
-	filterOpts, err := filter.FilterOptionsFromRequest(req, "id", apitype.SortDescending)
+	q, err = filter.FilterableDBResult(q, filterOpts, nil)
 	if err != nil {
-		RespondWithJSON(http.StatusInternalServerError, w, map[string]interface{}{"code": http.StatusInternalServerError,
-			"message": "Error building job run report:" + err.Error()})
-		return
+		return jobRuns, err
 	}
-	q, err = filter.FilterableDBResult(q, filterOpts, nil)
+
+	res := q.Find(&jobRuns)
+	return jobRuns, res.Error
+}
+
+// GetPayloadAnalysis loads the most recent payloads  for a stream and attempts to search for most commonly
+// failing tests, possible perma-failing blockers, etc.
+func GetPayloadAnalysis(dbc *db.DB, release, stream, arch string) (*apitype.PayloadStreamAnalysis, error) {
```

Comment:
I'm drifting towards calling these "payloads" not release tags, as I feel like release is a bit too overloaded and payload is the term we tend to use. Let me know what you think, I'd ultimately like to rename the tables as well if this sounds good to you as well.